bigframes/core/blocks.py

```diff
@@ -210,8 +211,7 @@ def shape(self) -> typing.Tuple[int, int]:
             except Exception:
                 pass
 
-        iter, _ = self.session._execute(row_count_expr, ordered=False)
-        row_count = next(iter)[0]
+        row_count = self.session._executor.get_row_count(self.expr)
```

Comment:
How do we decide what operations warrant a new method on the executor?

Reply:
Its a bit difficult to draw the line, but executor captures differents intents for consuming the dataframe, while nodes will represent value transformations not necessarily associated with immediate intent to consume. I think maybe some of this head stuff would be better handled with a bfet-level slice operator, but we don't have that at present.